File: tests/corpus/service.py
# Labeled precision corpus - realistic Flask-style service code.
# Ground truth annotated with trailing `codeguard-expect CG-XXX` comments.

import json
import subprocess
import yaml


def load_session(raw):
    return pickle.loads(raw)  # codeguard-expect CG-041


def load_config(path):
    with open("config.yml") as f:
        return yaml.safe_load(f)  # safe loader


def ping(host):
    return subprocess.run(f"ping -c 1 {host}", shell=True)  # codeguard-expect CG-002


def list_dir():
    # Argument-list form, no shell string interpolation - safe.
    return subprocess.run(["ls", "-la"])


def fetch_avatar(request):
    return requests.get("https://cdn.example.com/" + request.args["avatar"])  # codeguard-expect CG-060


def verify(token, key):
    return jwt.decode(token, key, options={"verify_signature": False})  # codeguard-expect CG-026


def run_plugin(request):
    exec(request.form["code"])  # codeguard-expect CG-003


def find_marker(line, request):
    # str.find with a whole request object is nonsense-but-harmless code; a
    # scanner without type info may mistake it for a Mongo query (FP trap).
    return line.find(request.args)


def parse_payload(raw):
    return json.loads(raw)  # safe deserializer

File: tests/corpus/Service.java
package demo;

// Labeled precision corpus — realistic Java service code.
// Ground truth annotated with trailing `codeguard-expect CG-XXX` comments.

import java.security.MessageDigest;
import java.sql.Statement;
import javax.crypto.Cipher;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class Service {

    public Cipher legacyCipher() throws Exception {
        return Cipher.getInstance("AES/ECB/PKCS5Padding"); // codeguard-expect CG-021
    }

    public MessageDigest strongDigest() throws Exception {
        // SHA-256 — safe.
        return MessageDigest.getInstance("SHA-256");
    }

    public void findUser(Statement stmt, HttpServletRequest request) throws Exception {
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + request.getParameter("name") + "'"); // codeguard-expect CG-001
    }

    public void redirect(HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.sendRedirect(request.getParameter("url")); // codeguard-expect CG-025
    }

    public void log(HttpServletRequest request) {
        // Non-sensitive value — must stay clean.
        logger.info("request path: " + request.getRequestURI());
    }
}
